packages/remirror__extension-entity-reference/src/entity-reference-extension.ts

```diff
@@ -0,0 +1,264 @@
+      }
+
+      const singleMark = this.type.create({ id: entityReferenceId });
+      try {
+        // Remove all entityReference marks with id=entityReferenceId in all the document
+        // Getting the last position https://discuss.prosemirror.net/t/getting-the-last-node-and-its-pos-in-the-document/3091
+        tr = tr.removeMark(1, tr.doc.content.size, singleMark);
+      } catch (error: any) {
+        throw new Error(`Can't remove entityReference ${entityReferenceId}: ${error.message}`);
+      }
+      dispatch?.(tr);
+      return true;
+    };
+  }
+
+  @command()
+  redrawEntityReferences(): CommandFunction {
+    return ({ tr, dispatch }) => {
+      dispatch?.(
+        tr.setMeta(EntityReferenceExtension.name, {
+          type: ActionType.REDRAW_ENTITY_REFERENCES,
+        }),
+      );
+
+      return true;
+    };
+  }
+
+  /**
+   * Get all disjoined entityReference attributes from the document.
+   */
+  @helper()
+  getDisjoinedEntityReferences(doc: ProsemirrorNode): Helper<EntityReferenceMetaData[][]> {
+    const disjoinedEntityReferences: EntityReferenceMetaData[][] = [];
+    doc.descendants((node: Node, pos: number) => {
+      const subDisjoinedEntityReferences = getDisjoinedEntityReferencesFromNode(
+        node,
+        pos,
+        this.name,
+      );
+
+      if (subDisjoinedEntityReferences.length === 0) {
+        return true;
+      }
+
+      disjoinedEntityReferences.push(subDisjoinedEntityReferences);
+      return true;
+    });
+    return disjoinedEntityReferences;
+  }
+
+  /**
+   * Disjoined entityReferences are analog to ProseMirror's marks. When adding a mark
+   * that spans two (or more) nodes (like paragraphs), it will be stored as two
+   * (or more) marks on each node separately. The same is true for disjoined
+   * entityReferences. Therefore, a entityReference that spans multiple paragraphs is
+   * internally stored as multiple marks (all having the same entityReference ID as
+   * attribute).
+   *
+   * To get the actual entityReferences (for which each entityReference ID is unique), call
+   * `joinDisjoinedEntityReferences`.
+   */
+  @helper()
+  getEntityReferences(): Helper<EntityReferenceMetaData[]> {
+    const entityReferences = getEntityReferencesFromPluginState({
+      extension: this,
+      state: this.store.getState(),
+    }).flat();
+    return joinDisjoinedEntityReferences(entityReferences);
+  }
+
+  /**
+   * @param pos - the position in the root document to find entityReference marks.
+   * @param includeEdges - whether to match entityReferences that start or end exactly on the given pos
+   *
+   * @returns all entityReferences at a specific position in the editor.
+   *
+   * In order to use this helper make sure you have the
+   * [[`EntityReferenceExtension`]] added to your editor.
+   */
+  @helper()
+  getEntityReferencesAt(
+    pos?: PrimitiveSelection,
+    includeEdges = true,
```

Comment:
I think we should drop this option still

Reply:
Good point. We don't need the config option ourselves, so will remove it for now to keep the extension simple.